### PR TITLE
Feat: Create contributor model

### PR DESCRIPTION
**Pull request by aurlic**
### Changes

523e58639d137179e7950145d87f578e86b6a2ff:
- added a new contributor model with all required fields

f8b62e1b3681d7ffde1c5fc3d2993e012cc0afd1:
- added a utility function used in view to check if a user is a project lead

c6eb6ae6f02cd92cb1d792c6ba03d2b43bb59a03:
- added a view for contributors, making notes and rank only visible by project-leads

### CODE REVIEW

**Comment by aurlic** [comment]
I have added the code for the Contributor model and for the view, but for some reason I can't reach localhost:8000 even after following the SETUP.md

**Comment by iamnovichek** [comment]
> I have added the code for the Contributor model and for the view, but for some reason I can't reach localhost:8000 even after following the SETUP.md 

Pls contact me on Telegram, we will find the solution together- @bobik300

File: tracker/models.py [hunk 1]
```diff
@@ -223,3 +229,26 @@ def create_approval_task(
...
+    user = models.ForeignKey(CustomUser, on_delete=models.CASCADE, related_name="contributors")
+    role = models.CharField(max_length=ROLE_MAX_CHARACTER_LENGTH, choices=Roles.choices, default=Roles.CONTRIBUTOR)
+    notes = models.TextField(blank=True, null=True)
+    rank = models.IntegerField(default=0)
+
+    class Meta:
+        verbose_name_plural = "Contributors"
+    
+    def __str__(self):
+        return f"Contributor: {self.user.email} (Role: {self.role})"
```
Comment (djeck1432) [inline tracker/models.py]: `user.email` is not so informative, I would rather use telegram_id or github_handler

File: tracker/views.py [hunk 2]
```diff
@@ -43,3 +45,36 @@ def post(self, request, *args, **kwargs) -> HttpResponse:
...
 
         return render(request, self.template_name, {"form": form})
+
+
+class ContributorListView(View):
+    """
+    View to list contributors. Only project leads can view notes and rank
+    """
+    
+    def get(self, request, *args, **kwargs):
```
Comment (djeck1432) [inline tracker/views.py]: add docstring

File: tracker/views.py [hunk 3]
```diff
@@ -43,3 +45,36 @@ def post(self, request, *args, **kwargs) -> HttpResponse:
         [messages.error(request, error_) for error_ in form.errors.values()]
 
         return render(request, self.template_name, {"form": form})
+
+
+class ContributorListView(View):
```
Comment (djeck1432) [inline tracker/views.py]: Use `ListView` it's more appropriate class and add `get_queryset` and `render_to_response`

**Review by djeck1432 — CHANGES_REQUESTED** [review]

File: tracker/models.py (hunk 1 above)
Comment (aurlic) [reply tracker/models.py]: addressed in [7964236](https://github.com/KostCode-Organization/hackathon_bot/pull/44/commits/796423621978b719855c3d98299c6f58af249c6d)

File: tracker/views.py (hunk 2 above)
Comment (aurlic) [reply tracker/views.py]: addressed in [f77882f](https://github.com/KostCode-Organization/hackathon_bot/pull/44/commits/f77882f985a79235eff9ab7a9d8d1bd378ce89f0)

File: tracker/views.py (hunk 3 above)
Comment (aurlic) [reply tracker/views.py]: addressed in [5a753c0](https://github.com/KostCode-Organization/hackathon_bot/pull/44/commits/5a753c0f83ae3af67c925a787f325f6f36598714)

File: tracker/models.py [hunk 4]
```diff
@@ -121,6 +121,12 @@ def is_staff(self) -> models.BooleanField:
         :return: bool
         """
         return self.is_admin
+    
+    def is_project_lead(self):
```
Comment (iamnovichek) [inline tracker/models.py]: Pls add return type and update your docstring

File: tracker/views.py [hunk 5]
```diff
@@ -43,3 +45,52 @@ def post(self, request, *args, **kwargs) -> HttpResponse:
         [messages.error(request, error_) for error_ in form.errors.values()]
 
         return render(request, self.template_name, {"form": form})
+
+
+class ContributorListView(ListView):
```
Comment (iamnovichek) [inline tracker/views.py]: We will manage all contributors via admin site so please move your logic in `tracker/admin.py`

**Review by iamnovichek — CHANGES_REQUESTED** [review]

**Comment by iamnovichek** [comment]
@aurlic pls sync with main branch

File: tracker/models.py (hunk 4 above)
Comment (aurlic) [reply tracker/models.py]: addressed in [c3d18b1](https://github.com/KostCode-Organization/hackathon_bot/pull/44/commits/c3d18b1846618fec4417236ec21a94fb4237172a)

File: tracker/views.py (hunk 5 above)
Comment (aurlic) [reply tracker/views.py]: addressed in [ea8a970](https://github.com/KostCode-Organization/hackathon_bot/pull/44/commits/ea8a970dbdf23b0c8fabf29e52fdd189d3ff6fe6)

**Comment by iamnovichek** [comment]
@aurlic pls sync with main

**Comment by aurlic** [comment]
Here is a screenshot showing that the Contributor model works and that the visibility logic does aswell.
2024_12_01_0fv_Kleki

**Review by iamnovichek — CHANGES_REQUESTED** [review]

**Review by iamnovichek — APPROVED** [review]

**Pull request merged**